### PR TITLE
Fix resource name when Grape paths contain a leading slash

### PR DESCRIPTION
**Pull request by jethrodaniel**
**What does this PR do?**

Fix an issue where a [Grape](https://github.com/ruby-grape/grape) resource like this

```ruby
class TestingAPI < Grape::API
  get '/path/with/leading/slash' do
    'leading-slash'
  end
end
```

results in a resource `TestingAPI GET //path/with/leading/slash`, when it _should_ result in `TestingAPI GET /path/with/leading/slash`.

**Motivation:**

Seeing the resource name have an unnecessary leading slash.

**Change log entry**

Fix resource name when Grape paths contain a leading slash

**Additional Notes:**

**How to test the change?**

```
rake test:grape
```

### CODE REVIEW

File: lib/datadog/tracing/contrib/grape/endpoint.rb [hunk 1]
```diff
@@ -268,7 +268,7 @@ def endpoint_expand_path(endpoint)
               namespace = endpoint.routes.first && endpoint.routes.first.namespace || ''
 
               parts = (namespace.split('/') + route_path).reject { |p| p.blank? || p.eql?('/') }
-              parts.join('/').prepend('/')
+              parts.join('/').delete_prefix('/').prepend('/')
```
Comment (Strech) [inline lib/datadog/tracing/contrib/grape/endpoint.rb:271]: Hey @jethrodaniel thanks for your contribution 👍🏼 

I think given that we already have a parts with a first part to be prefixed, adding and removing `/` looks excessive. WDYT if we just check before prepending if the first char is a `/`?

Something like:

```ruby
path = (namespace.split('/') + route_path).reject { |p| p.blank? || p.eql?('/') }.join('/')
path.prepend('/') if path[0] != '/'
```

Comment (jethrodaniel) [reply lib/datadog/tracing/contrib/grape/endpoint.rb:271]: +1 for the more efficient approach - will update when I get time. Thanks!

**Review by Strech — APPROVED** [review]
LGTM 👍🏼 

Thanks for your time 💟

**Review by p-datadog — APPROVED** [review]

**Comment by p-datadog** [comment]
We need to verify that `http.route` is the same as `resource_name` with respect to the leading slashes.

**Comment by Strech** [comment]
Hey @jethrodaniel thanks for your contribution. This change was released as a part of [v2.5.0](https://github.com/DataDog/dd-trace-rb/releases/tag/v2.5.0) release. 🎉

**Pull request merged**